packages/miniflare/src/workers/queues/broker.worker.ts

```diff
@@ -320,28 +340,36 @@ export class QueueBrokerObject extends MiniflareDurableObject<QueueBrokerObjectE
 		this.#pendingFlush = { immediate: delay === 0, timeout };
 	}
 
-	#enqueue(messages: QueueIncomingMessage[]) {
+	#enqueue(messages: QueueIncomingMessage[], globalDelay: number = 0) {
 		for (const message of messages) {
 			const randomness = crypto.getRandomValues(new Uint8Array(16));
 			const id = message.id ?? Buffer.from(randomness).toString("hex");
 			const timestamp = new Date(message.timestamp ?? this.timers.now());
 			const body = deserialise(message);
-			this.#messages.push(new QueueMessage(id, timestamp, body));
+			const msg = new QueueMessage(id, timestamp, body);
+
+			const fn = () => {
+				this.#messages.push(msg);
+				this.#ensurePendingFlush();
+			}
+
+			const delay = message.delaySecs ?? globalDelay;
+			this.timers.setTimeout(fn, delay * 1000);
 		}
-		this.#ensurePendingFlush();
```

Comment:
This is a change in behaviour from ensuring a flush at the end of all messages to ensureing a flush at the end of each message. Will this cause problems?